interface/wx/webrequest.h

```diff
@@ -0,0 +1,639 @@
+        <td>Windows</td>
+        <td>
+            <a target=_new href="https://docs.microsoft.com/en-us/windows/desktop/WinHttp/about-winhttp">WinHTTP</a>
+        </td>
+        <td>Yes</td>
+        <td>Windows 10 1607</td>
+    </tr>
+    <tr>
+        <td>macOS</td>
+        <td>
+            <a target=_new href="https://developer.apple.com/documentation/foundation/urlsession">NSURLSession</a>
+        </td>
+        <td>macOS 10.9</td>
+        <td>macOS 10.11</td>
+    </tr>
+    <tr>
+        <td>iOS</td>
+        <td>
+            <a target=_new href="https://developer.apple.com/documentation/foundation/urlsession">NSURLSession</a>
+        </td>
+        <td>iOS 7.0</td>
+        <td>iOS 9.0</td>
+    </tr>
+    <tr>
+        <td>Linux</td>
+        <td>
+            <a target=_new href="https://curl.haxx.se/libcurl/">libcurl</a>
+        </td>
+        <td>Yes</td>
+        <td>7.47.0</td>
+    </tr>
+    </table>
+
+    @beginEventEmissionTable{wxWebRequestEvent}
+    @event{wxEVT_WEBREQUEST_STATE(id, func)}
+        The request state changed.
+    @event{wxEVT_WEBREQUEST_DATA(id, func)}
+        A new block of data has been downloaded.
+    @endEventTable
+
+    @since 3.1.3
+
+    @library{wxnet}
+    @category{net}
+
+    @see wxWebResponse, wxWebSession
+*/
+class wxWebRequest: public wxEvtHandler, public wxRefCounter
+{
+public:
+    /**
+        Possible request states returned by GetState().
+    */
+    enum State
+    {
+        /// The request has just been created and Start() has not been called
+        State_Idle,
+
+        /**
+            The request is currently unauthorized. Calling GetAuthChallenge()
+            returns a challenge object with further details.
+        */
+        State_Unauthorized,
+
+        /// The request has been started and is transferring data
+        State_Active,
+
+        /// The request completed successfully and all data has been received.
+        State_Completed,
+
+        /// The request failed
+        State_Failed,
+
+        /// The request has been cancelled before completion by calling Cancel()
+        State_Cancelled
+    };
+
+    /**
+        Possible storage types. Set by SetStorage().
+    */
+    enum Storage
+    {
+        /// All data is collected in memory until the request is complete
+        Storage_Memory,
+
+        /// The data is written to a file on disk
+        Storage_File,
+
+        /**
+            The data is not stored by the request and is only available
+            via events.
+        */
+        Storage_None
+    };
+
+    /**
+        Send the request to the server asynchronously.
+
+        Events will be triggered on success or failure.
+
+        @see Cancel()
+    */
+    void Start();
+
+    /**
+        Cancel an active request.
+    */
+    void Cancel();
+
+    /**
+        Returns a response object after a successful request.
+
+        Before sending a request or after a failed request this will return
+        @c NULL.
+    */
```

Comment:
Please mention that the pointer is non-owning (as I think it is) for this and other methods returning raw pointers.